### PR TITLE
feat: update installer to podman '4.2.0', download mac packages

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Update installer to use podman `4.2.0`, also it add downloading podman packages for MacOS

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
https://github.com/containers/podman-desktop/issues/370

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
how does it work for mac installer and generated dmg files ?

do we include amd64.pkg only in the amd64 installer and aarch64.pkg in the M1 dmg ?

**Comment by evidolob** [comment]
We include both pkg for macos.
This PR add only packaging pkg files for mac, installation will be done in separate PR

**Comment by benoitf** [comment]
but we should include files only for the target dmg 
here it's adding `2 x 80MB`

if I download the amd64 DMG, I expect to have a smaller binary with only amd64 stuff inside, no aarch64 installer

**Comment by evidolob** [comment]
What I should do in case of `universal` binary?

**Comment by benoitf** [comment]
universal: include both (it duplicates size of intel + arm)

**Comment by evidolob** [comment]
@benoitf Now only one `pkg` will be packed in app. Universal app will contains both `pkg`.

File: .electron-builder.config.js [hunk 1]
```diff
@@ -36,8 +37,30 @@ const config = {
...
+      context.packager.config.extraResources.push('extensions/podman/builtin/podman.cdix/assets/podman-installer-macos-amd64-*.pkg');
+    }
+
+    if(context.arch === Arch.x64 && context.electronPlatformName === 'win32'){
+      context.packager.config.extraResources.push('extensions/podman/builtin/podman.cdix/assets/**');
+    }
+  },
   files: ['packages/**/dist/**', 'extensions/**/builtin/*.cdix/**'],
-  extraResources: ['packages/main/src/assets/**', 'extensions/podman/builtin/podman.cdix/assets/**'],
+  extraResources: [],
```
Comment (benoitf) [inline .electron-builder.config.js]: do we need to define extraResources if it's empty ?

**Comment by benoitf** [comment]
Hi @evidolob 

when I'm looking at CI jobs
before your PR we had for example on this job: https://github.com/containers/podman-desktop/actions/runs/2875251610#artifacts
image

so like around 90MB for mac specific arches and 157 MB for universal (delta is 67MB)

now, with your updated PR https://github.com/containers/podman-desktop/actions/runs/2880379602#artifacts
image

around 250MB for arches and 315MB for universal so we still have a delta of around 67MB but each arch has a growing size from 90MB to 249MB for arm64

but if I look at installers
image

pkg files are around 80MB

so we should have specif arches dmg growing of around 80MB
I expect: 90MB + 80MB = 170MB but here we see 250MB

so it seems it's still including two assets because 250MB-(original size of 90MB)-(pkg installer of 80MB)=80MB

**Review by benoitf — APPROVED** [review]

File: .electron-builder.config.js (hunk 1 above)
Comment (evidolob) [reply .electron-builder.config.js]: No

**Comment by benoitf** [comment]
size are the one expected

but when I launch Podman Desktop on macos, it's not asking to install podman ?

image

**Comment by evidolob** [comment]
Yes, as it not part of this PR, on this PR I add only packaging podman for macos, later PR will add installation.

**Pull request merged**